src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -0,0 +1,259 @@
+            return []
+        policies = json.loads(output)
+        yield from policies.get("PolicyNames", [])
+
+    def list_attached_policies_in_role(self, role_name: str):
+        list_attached_policies_cmd = (
+            f"aws iam list-attached-role-policies --profile {self._profile} --role-name {role_name} --no-paginate"
+        )
+        code, output, error = run_command(list_attached_policies_cmd)
+        if code != 0:
+            logger.error(error)
+            return []
+        policies = json.loads(output)
+        for policy in policies.get("AttachedPolicies", []):
+            yield policy.get("PolicyArn")
+
+    def get_role_policy(self, role_name, policy_name: str | None = None, attached_policy_arn: str | None = None):
+        if policy_name:
+            get_policy = (
+                f"aws iam get-role-policy --profile {self._profile} --role-name {role_name} "
+                f"--policy-name {policy_name} --no-paginate"
+            )
+        elif attached_policy_arn:
+            get_attached_policy = (
+                f"aws iam get-policy --profile {self._profile} --policy-arn {attached_policy_arn} --no-paginate"
+            )
+            code, output, error = run_command(get_attached_policy)
+            if code != 0:
+                logger.error(error)
+                return []
+            policy_version = json.loads(output)["Policy"]["DefaultVersionId"]
+            get_policy = (
+                f"aws iam get-policy-version --profile {self._profile} --policy-arn {attached_policy_arn} "
+                f"--version-id {policy_version} --no-paginate"
+            )
+        else:
+            logger.error("Failed to retrieve role. No role name or attached role ARN specified.")
+            return []
+        code, output, error = run_command(get_policy)
+        if code != 0:
+            logger.error(error)
+            return []
+        policy = json.loads(output)
+        if policy_name:
+            actions = policy["PolicyDocument"].get("Statement", [])
+        else:
+            actions = policy["PolicyVersion"]["Document"].get("Statement", [])
+        for action in actions:
+            if action.get("Effect", "Deny") != "Allow":
+                continue
+            actions = action["Action"]
+            s3_action = []
+            if isinstance(actions, list):
+                for single_action in actions:
+                    if single_action in self.S3_ACTIONS:
+                        s3_action.append(single_action)
+                        continue
+            elif actions in self.S3_ACTIONS:
+                s3_action = [actions]
+
+            if not s3_action:
+                continue
+            for resource in action.get("Resource", []):
+                match = re.match(self.S3_REGEX, resource)
+                if match:
+                    yield AWSPolicyAction("s3", set(s3_action), match.group(1))
+
+
+class AWSInstanceProfileCrawler(CrawlerBase[AWSInstanceProfile]):
```

Comment:
it's not a crawler. we don't need to display this in a dashboard. . make CSV mapping, same way we do for azure

Reply:
Fixed